buildtools/commands/test.js

```diff
@@ -0,0 +1,73 @@
+const { resolve } = require('path');
+const { execSync } = require('child_process');
+
+exports.command = 'test';
+exports.desc = 'Runs tests for the Okta Sign-in Widget';
+exports.builder = {
+  type: {
+    description: 'Type of the test runner',
+    choices: [
+      'karma',
+      'jest',
+    ],
+    required: true,
+  },
+  config: {
+    description: 'Optional override for the test config file',
+  },
+  suiteHelp: {
+    description: 'Run this to see the test CLI options',
+  },
+};
+
+const suiteMap = {
+  karma: {
+    bin: 'karma start',
+    config: 'karma.conf.js',
+    preReq: [
+      'grunt assets',
+    ],
+  },
+  jest: {
+    bin: 'jest',
+    config: 'jest.config.js',
+    preReq: [
+      'grunt assets',
+    ],
+  },
+};
+
+exports.handler = async (argv) => {
+  const packageDir = process.cwd();
+  const testType = suiteMap[argv.type];
+  
+  // Start building the test command
+  let cmd = testType.bin;
+
+  if (testType.config) {
+    // If we have an available configuration file, use it
+    const defaultConfig = resolve(packageDir, testType.config);
+    cmd += argv.config ? '' : ` --config ${defaultConfig}`;
+  }
+
+  // Show CLI configuration
+  const help = argv.suiteHelp ? ' --help' : '';
+
+  // Get additional CLI args
+  const additionalArgs = process.argv.slice(5).join(' ');
```

Comment:
What is an example of `addtionalArgs` ? Is it specifying a testFile name to run against?

Reply:
The `additionalArgs` is mapped to all CLI configuration flags for both [jest](https://jestjs.io/docs/cli) and [karma](http://karma-runner.github.io/6.3/config/configuration-file.html).

The idea is to let developers run the `--suiteHelp` command to output a list of **all available CLI options**:

#### yarn test --type jest --suiteHelp
```sh
Running: jest --config /${path}/jest.config.js --help --suiteHelp
Usage: jest [--config=<pathToConfigFile>] [TestPathPattern]

Options:
  --help, -h                    Show help                              [boolean]
  --version, -v                 Print the version and exit             [boolean]
  --all                         The opposite of `onlyChanged`. If `onlyChanged`
                                is set by default, running jest with `--all`
                                will force Jest to run all tests instead of
                                running only tests related to changed files.
                                                                       [boolean]
  --automock                    Automock all files by default.         [boolean]
  --bail, -b                    Exit the test suite immediately after `n` number
                                of failing tests.                      [boolean]
  --browser                     Respect the "browser" field in package.json when
                                resolving modules. Some packages export
                                different versions based on whether they are
                                operating in node.js or a browser.     [boolean]
  --cache                       Whether to use the transform cache. Disable the
                                cache using --no-cache.                [boolean]
  --cacheDirectory              The directory where Jest should store its cached
                                dependency information.                 [string]
...
```

Then, a developer can pick and choose which CLI args to pass:

```sh
# Generate coverage report
yarn test --type jest --coverage

# Run jest in debug mode
yarn test --type jest --debug
```